packages/liveblocks-react-lexical/src/comments/threads-panel.tsx

```diff
@@ -0,0 +1,429 @@
+
+    observer.observe(root);
+    return () => observer.disconnect();
+  }, [root, handlePositionThreads]);
+
+  if (orderedThreads.length === 0) return;
+
+  return (
+    <div
+      {...divProps}
+      className={classNames(className, "lb-root lb-lexical-thread-panel")}
+      ref={containerRef}
+      style={{
+        position: "relative",
+        ...style,
+      }}
+    >
+      {orderedThreads.map(({ thread, rect }) => {
+        let top = rect.top;
+
+        if (positions.has(thread.id)) {
+          top = positions.get(thread.id)!;
+        }
+
+        const isActive = activeThreads.includes(thread.id);
+
+        return (
+          <ThreadWrapper
+            key={thread.id}
+            Thread={Thread}
+            thread={thread}
+            onItemAdd={onItemAdd}
+            onItemRemove={onItemRemove}
+            style={{
+              position: "absolute",
+              transform: `translateX(${isActive ? -12 : 0}px) translateY(${top}px)`,
+              left: 0,
+              width: "100%",
+              paddingBottom: gap,
+              transition: "transform 0.2s",
+            }}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+interface ThreadWrapperProps extends ThreadProps {
+  Thread: ComponentType<ThreadProps>;
+}
+
+interface ThreadWrapperProps extends HTMLAttributes<HTMLDivElement> {
+  onItemAdd: (id: string, el: HTMLElement) => void;
+  onItemRemove: (id: string) => void;
+}
+
+function ThreadWrapper({
+  onItemAdd,
+  onItemRemove,
+  thread,
+  Thread,
+  ...divProps
+}: ThreadWrapperProps) {
+  const [editor] = useLexicalComposerContext();
+  const nodes = useThreadToNodes();
+  const divRef = useRef<HTMLDivElement>(null);
+
+  const activeThreads = useActiveThreads();
+
+  const isActive = activeThreads.includes(thread.id);
+
+  function handleThreadClick() {
+    const keys = nodes.get(thread.id);
+    if (keys === undefined || keys.size === 0) return;
+
+    if (activeThreads.includes(thread.id)) return;
+
+    editor.update(
+      () => {
+        const [key] = keys;
+        const node = $getNodeByKey(key);
+        if (!$isThreadMarkNode(node)) return;
+        node.selectStart();
+      },
+      {
+        onUpdate: () => {
+          const container = divRef.current;
+          if (container === null) return;
+
+          const composer = container.querySelector(".lb-composer-editor");
+          if (composer !== null) {
+            if (!(composer instanceof HTMLElement)) return;
+            composer.focus();
+          }
+          // If the composer is not found (and this can happen if `showComposer` is set to `false`), wait for the next render and try again to focus the composer.
+          else {
+            setTimeout(() => {
+              const composer = container.querySelector(".lb-composer-editor");
+              if (composer === null) return;
+              if (!(composer instanceof HTMLElement)) return;
+              composer.focus();
```

Comment:
After trying it a bit, I feel like this has too many cases where I would prefer to not focus the composer on active, just opening it.